components/shopify/actions/create-order/create-order.mjs

```diff
@@ -0,0 +1,142 @@
+import shopify from "../../shopify.app.mjs";
+import { toSingleLineString } from "../commons.mjs";
+
+export default {
+  key: "shopify-create-order",
+  name: "Create Order",
+  description: toSingleLineString(`Creates a new order.
+    For full order object details [see the docs](https://shopify.dev/api/admin-rest/2022-01/resources/order#[post]/admin/api/2022-01/orders.json)
+    or [see examples](https://shopify.dev/api/admin-rest/2022-01/resources/order#[post]/admin/api/#{api_version}/orders.json_examples)
+  `),
+  version: "0.0.1",
+  type: "action",
+  props: {
+    shopify,
+    lineItems: {
+      type: "string[]",
+      label: "Line Items",
+      description: toSingleLineString(`
+        A list of line item objects, each containing information about an item in the order.
+        More details when searching **line_items** in [Shopify Order Object](https://shopify.dev/api/admin-rest/2022-01/resources/order#resource_object)
+      `),
+    },
+    customerId: {
+      propDefinition: [
+        shopify,
+        "customerId",
+      ],
+      optional: true,
+    },
+    billingAddress: {
+      type: "object",
+      label: "Billing Address",
+      description: toSingleLineString(`
+        The mailing address associated with the payment method.
+        More details when searching **billing_address** in [Shopify Order Object](https://shopify.dev/api/admin-rest/2022-01/resources/order#resource_object)
+      `),
+      optional: true,
+    },
+    shippingAddress: {
+      type: "object",
+      label: "Shipping Address",
+      description: toSingleLineString(`
+        The mailing address to where the order will be shipped.
+        More details when searching **billing_address** in [Shopify Order Object](https://shopify.dev/api/admin-rest/2022-01/resources/order#resource_object)
+      `),
+      optional: true,
+    },
+    financialStatus: {
+      type: "string",
+      label: "Financial Status",
+      description: "The status of payments associated with the order. Can only be set when the order is created",
+      options: [
+        "pending",
+        "authorized",
+        "partially_paid",
+        "paid",
+        "partially_refunded",
+        "refunded",
+        "voided",
+      ],
+      optional: true,
+    },
+    discountCodes: {
+      type: "string[]",
+      label: "Discount Codes",
+      description: toSingleLineString(`
+        A list of discounts applied to the order.
+        Example: \`[ { "code": "SPRING30", "type": "fixed_amount", "amount": "30.00" } ]\`.
+        More details when searching **discount_codes** in [Shopify Order Object](https://shopify.dev/api/admin-rest/2022-01/resources/order#resource_object)
+      `),
```

Comment:
<img width="400" alt="discount_codes" src="https://user-images.githubusercontent.com/19861096/153686066-c8bab264-ec47-4093-86e7-bc7e2ebd9bae.png">

When I enter `[ { "code": "SPRING30", "type": "fixed_amount", "amount": "30.00" } ]` as this prop's value, I get the error:

```
string type not supported, please use array of objects in structured mode
```

Entering a custom expression in `{{}}` works as a [non-structured param](https://pipedream.com/docs/workflows/steps/params/#structured-params):
```
{{[ { "code": "SPRING30", "type": "fixed_amount", "amount": "30.00" } ]}}
```

For this prop and other "**array of JSON**" props, do you think we should
- change the example in the description to include `{{}}`
- or show just one array item (e.g. `{ "code": "SPRING30", "type": "fixed_amount", "amount": "30.00" }`), used in a structured param
<img width="350" alt="discount_structured" src="https://user-images.githubusercontent.com/19861096/153687081-6e9575fc-170a-4b08-8ae9-88bbfee803b7.png">

- or update [`parseArrayOfJSONStrings()`](https://github.com/PipedreamHQ/pipedream/blob/8788d99857f68082d5093cdaeefcf3b69424c50d/components/shopify/shopify.app.mjs#L617) to support string parsing?

Reply:
Changed Discount Codes to one Discount Code object. I think the idea of simplifying props is better. Let me know what do you think.